### DISCUSSION TITLE
RaspberryMatic 3.57.5.20210424

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#1236] [Announcements]
This is release 3.57.5.20210424 of RaspberryMatic which is a **maintenance** release with bugfixes and feature changes including the following changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.57.5.20210424/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.57.5.20210424) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=67349)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.57.4.20210320...3.57.5.20210424).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.57.5-1](https://github.com/jens-maus/occu/tree/b_3_57) integrating full compatibility to the [CCU3 3.57.5 firmware](https://www.eq-3.de/downloads/software/firmware/ccu3-firmware/CCU3-Changelog.3.57.5.pdf) which comes with the following changes:
  - after device reset config data was not correctly transmitted to HomeMatic/homematicIP devices anymore.
  - when editing device configuration values where devices accepted value < 0 ended up in not being able to enter 0 anymore.
  - when adding channel 1 of a HmIP(W)-FAL device also channel 2 was accidentally added in the direct channel connection which did not have any function.
- modified the "Clone System" operation in the recovery system to also deal with clone operations where the target device is smaller than the source device. In this case the user partition will be created smaller before using rsync to copy the userfs data from source to destination.
- fixed the firmware update/install routines in the recovery system to write the bootloader for any partitioning scheme where the boot partition starts >= 2048.
- integrated a minor fix which could have lead to an empty `HM_HMIP_ADDRESS_ACTIVE` variable entry in `/var/hm_mode` in case no `/etc/config/hmip_address.conf` file exists upon bootup.

#### WebUI changes:
- changed 0119-WebUI-Fix-ProofAndSetValue WebUI bugfix patch with more fixes for the `ProofAndSetValue` function not correctly allowing to deal with all device relevant value boundaries in the WebUI device configuration dialogs (#1231, @theimo1221, @Baxxy13).
- integrated a fix in the 0111-WebUI-Fix-Translation-Performance WebUI patch which should fix issues where empty translation strings resulted  issue that which should finally fix #1155.
- modified 0060-WebUI-Fix-SessionLogoutSave WebUI patch to not patch the `Session.logout()` JSON RPC to use `system.ClearSession()` rather than `system.ClearSessionID()` because using the non-ID version causes the regadom to be saved under all circumstances and for the remote API this might not be something we actually want (#1192).

#### Operating system changes:
- added hardware support for the [Hardkernel ODROID](https://www.hardkernel.com/) line of SBC platform ([ODROID-N2/N2+](https://wiki.odroid.com/odroid-n2/odroid-n2), [ODROID-C4](https://wiki.odroid.com/odroid-c4/odroid-c4), [ODROID-C2](https://wiki.odroid.com/odroid-c2/odroid-c2))
- added hardware support for the Intel NUC Gen 11 (Panther Canyon) hardware.
- integrated support for evaluating a global `/etc/sysctl.conf` file to tune certain linux system/kernel parameters.
- moved the `fstrim` cronjob to run at 04:00 each saturday and the badblock check to 01:59 to make sure it is run even during DST switch times. This closes #1204.
- updated `rpi-eeprom` package to incorporate latest `pieeprom-2021-03-18.bin` RaspberryPi4 EEPROM fixing potential issues with older pi4 board revisions.
- integrated certain fixes and enhancements for running RaspberryMatic as a full-fledged HomeAssistant Add-on.
- when running as a HomeAssistant Add-on we don't use `udev` anymore because the host OS is running its' own `udev` which could conflict otherwise.
- on the corresponding platforms we reset the board upon an U-Boot boot failure so that systems won't stall in the bootloader.
- updated `generic_raw_uart` to latest master version and removed our manual GPIO reset routine in the `InitRFHardware` init scripts because the newer `generic_raw_uart` handles the different reset pin layout of the `RPI-RF-MOD` vs. `HM-MOD-RPI-PCB` modules internally using a dedicated 'alt_reset_pin' device tree node entry now.
 - removed the dedicated `rx8130` buildroot package in favour of using the kernel mainline `ds1307,rx8130` module which since 5.10.x supports supercap charging (cf. https://lore.kernel.org/linux-rtc/20200917183246.19446-5-bst@pengutronix.de/T/).
- bumped `lighttpd` web server to latest stable 1.4.59 which fixes an issue with lengthy request field sizes resulting in not being able to updated HomeMatic device parameters (cf. https://redmine.lighttpd.net/issues/3059, #1189).
- bumped buildroot version to latest 2021.02.1 version.
- bumped 5.10.x LTS kernel to 5.10.32 for all appropriate platforms (Tinkerboard, ODROID, intelNUC, OVA).
- bumped 5.10.x LTS kernel for RaspberryPi to latest 1.20210303-1 version including the corresponding bootloader/firmware files.
- bumped U-Boot version to latest 2021.04 version including u-boot related patch updates for the tinkerboard target.
- bumped JAVA version to latest 8.54.0.21-ca-jdk8.0.292 LTS for appropriate platforms.

## :family: Contributors (alphabetically):
- @Baxxy13
- @jens-maus
- @theimo1221

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi4 Compute Module, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-rpi4.zip)<br/>SHA256: 612c0876b9a27b4cfebaf76cf4c3e5454290b91ce24be17f6f428b1d77592492</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-rpi3.zip)<br/>SHA256: a2028450f2a37040a7b61bba3497f3e38ce999c9c676e6ce2e2b376b8a6be8db</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: c20b74ec0609ef9755f149f53f5fb85c028fccb38fe14cf736c458888bf702d6</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-rpi2.zip)<br/>SHA256: 3df07bf788af2719d5a7d0ed86687c1efc46883915eab435d44060e1554265ab</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-rpi0.zip)<br/>SHA256: b7b9bb58e02d7e1d05fc2af5e88ed8c227693ee4f19614ad2076f9e35a146731</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-tinkerboard.zip)<br/>SHA256: 2a1d0f3ef55535094bd9482d4926349d8bfa6ee54310939b49745737203c17af</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-odroid-n2.zip)<br/>SHA256: c3ec74d5886f0482883d0a573d55e053c6b07f4ddda09b3d987945473170b6df</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-odroid-c4.zip)<br/>SHA256: 67aa265803cf21f30552cab6e1e04a92a95617cb93decbaff503cf2d96b61080</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-odroid-c2.zip)<br/>SHA256: 4cb1ea0d3412272a6b973dce24d9dfefb6a26d2141c3ccba22d18d7ce21dea08</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-intelnuc.zip)<br/>SHA256: 5686b2aca14986e8c8d180a23d1be0253dc443b8e334cf27c4f5c0560f5901ed</sub>

- Open Virtual Appliance (OVA) – ([ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Proxmox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-ova.zip)<br/>SHA256: f55b804904cc712b149093bcc8b4f7d828722decb85206876614a6af28d37415</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424.ova) (only for initial OVA installation)<br/>SHA256: d456ed97c454a49dd60c764275cba475148a4a69c6b0c0171c130adb1501f615</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.57.5.20210424-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 59b8ed8be42dc1a3f1be9977ddd54717f4de56fa2fd4dfce21c9b898ed186692</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: d497120bcbc4ee3cda11ee26f915c03326e760458f1a0d4b1083c6775573e409</sub>
<sub>:package: [RaspberryMatic-3.57.5.20210424-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.57.5.20210424/RaspberryMatic-3.57.5.20210424-oci_arm.tgz) (arm/armhf)<br/>SHA256: e73851de00a8c7b880f7731f35105a18b8ba4e39c21ac3febc860c0d4d034cc1</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.57.5.20210424'>RaspberryMatic 3.57.5.20210424</a>.</em>

**Comment by Madaraxa** [comment]
Hi,
Danke für das Update. Ich warte sehnsüchtig, da ich hiermit das Wochenprogramm wieder übertragen kann wenn ich es richtig verstanden habe. Aber gibt es einen Grund, warum es kein rpi2 image gibt? Wenn ich den link öffne kommt die Meldung "not found". 
Bis "eben" (vor ein paar Stunden) wurde mir das Update auch auf meinem RPi2 raspberrymatic angezeigt. Auch diese Meldung ist jetzt weg.

**Reply by Baxxy13** [reply]
Das wurde offline genommen weil es augenscheinlich fehlerhaft ist. Damit booteten die Pi2B's nicht. 
Siehe [Homematic-Forum](https://homematic-forum.de/forum/viewtopic.php?f=65&t=67349&start=10#p659301)

**Reply by Madaraxa** [reply]
Danke :)

**Reply by jens-maus** [reply]
Wenn ich es schaffe und der Fehler geringer Natur ist sollte es heute im Laufe des Tages wieder online gehen.

**Reply by jens-maus** [reply]
Inzwischen sollte das RaspberryPi2 und RaspberryPi0 Image wieder online sein und nun korrekt booten und funktionieren.
